src/main/java/edu/harvard/iq/dataverse/harvest/server/web/servlet/OAIServlet.java

```diff
@@ -233,150 +232,30 @@ private void processRequest(HttpServletRequest request, HttpServletResponse resp
                         "Sorry. OAI Service is disabled on this Dataverse node.");
                 return;
             }
+                        
+            RawRequest rawRequest = RequestBuilder.buildRawRequest(httpServletRequest.getParameterMap());
             
-            OAIRequestParametersBuilder parametersBuilder = newXoaiRequest();
-            
-            for (Object p : request.getParameterMap().keySet()) {
-                String parameterName = (String)p; 
-                String parameterValue = request.getParameter(parameterName);
-                parametersBuilder = parametersBuilder.with(parameterName, parameterValue);
-
-            }
-            
-            OAIPMH handle = dataProvider.handle(parametersBuilder);
+            OAIPMH handle = dataProvider.handle(rawRequest);
             response.setContentType("text/xml;charset=UTF-8");
-            
-            if (isGetRecord(request) && !handle.hasErrors()) {
-                writeGetRecord(response, handle);
-            } else if (isListRecords(request) && !handle.hasErrors()) {
-                writeListRecords(response, handle);
-            } else {
-                XmlWriter xmlWriter = new XmlWriter(response.getOutputStream());
-                xmlWriter.write(handle);
-                xmlWriter.flush();
-                xmlWriter.close();
-            }
+
+            XmlWriter xmlWriter = new XmlWriter(response.getOutputStream(), repositoryConfiguration);
+            xmlWriter.write(handle);
+            xmlWriter.flush();
+            xmlWriter.close();
```

Comment:
**suggestion:** You might consider creating the writer within a try-with-resources to avoid any spilled beans.